### PR TITLE
Linalg lowering for aten.conv2d and aten.AdaptiveAvgPool2d

### PR DESCRIPTION
**Pull request by cathyzhyi**
1. Add m_TorchConstantIntList
2. Lowering for aten.conv2d
3. Lowering aten.AdaptiveAvgPool2d

### CODE REVIEW

**Comment by cathyzhyi** [comment]
this change is blocked by upstream issues so can't be merged yet

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 1]
```diff
@@ -59,6 +59,245 @@ static LogicalResult verifyLinalgCompatibleTypes(Operation *op,
   return success();
 }
 
+// Hack to deal with the Torch list type arguments which is not supported end
+// to end. Constant values can be be extracted directly and non constant
+// list values are not supported.
+// TODO: loose this constraint when properly support list type
+static bool constantIntListAsExpected(Value &value,
```
Comment (silvasean) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:66]: can you give this a name that matches the coding standard? https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

I would call it `isConstantIntListMatching(Value &v, ArrayRef<int64_t> expected)`

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 2]
```diff
@@ -59,6 +59,245 @@ static LogicalResult verifyLinalgCompatibleTypes(Operation *op,
...
+    Value initTensor0 =
+        rewriter.create<linalg::FillOp>(loc, c0float, initTensor).getResult(0);
+
+    Value conv2d =
+        rewriter
+            .create<linalg::ConvNCHWOp>(loc, ranked4DTensorType,
+                                        ValueRange{paddedInput, weight},
+                                        ValueRange{initTensor0})
+            .getResult(0);
+    rewriter.replaceOp(op, conv2d);
```
Comment (silvasean) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:295]: do you need a tensor.cast op here?

**Review by silvasean — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 2 above)
Comment (cathyzhyi) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:295]: Right. Thanks for catching this!

**Comment by silvasean** [comment]
Is this still WIP? It still says WIP in the title an there are some remaining changes requested.

**Comment by cathyzhyi** [comment]
> Is this still WIP? It still says WIP in the title an there are some remaining changes requested.

I removed the WIP and didn't find a proper way to block merging.
I can see 2 comments on my side and they are all addressed.

**Review by silvasean — APPROVED** [review]

**Pull request merged**